### PR TITLE
Opera: Fix checkver script and update version to 104.0.4944.54

### PR DESCRIPTION
**Pull request by gigabotan**
Fix Opera checkver script to get latest version properly.
Bump version to 104.0.4944.54

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/opera.json [hunk 1]
```diff
@@ -35,17 +35,23 @@
     ],
     "persist": "profile",
     "checkver": {
-        "url": "http://get.opera.com/ftp/pub/opera/desktop/",
-        "regex": "([\\d.]+)/",
-        "reverse": true
+        "script": [
```
Comment (jasongodev) [inline bucket/opera.json:38]: Why do we need a script when a simple reversed regex result works fine? Did I missed something?

File: bucket/opera.json [hunk 2]
```diff
@@ -35,17 +35,23 @@
     ],
     "persist": "profile",
     "checkver": {
-        "url": "http://get.opera.com/ftp/pub/opera/desktop/",
```
Comment (jasongodev) [inline bucket/opera.json:38]: We can update this into: https://get.geo.opera.com/pub/opera/desktop/

File: bucket/opera.json (hunk 1 above)
Comment (gigabotan) [reply bucket/opera.json:38]: Regex returns first match on the page, but versions are not properly sorted on the FTP page.

**Comment by danielpauldd** [comment]
I have tested the manifest and it works perfectly.
The PR would also fix #12628

**Comment by danielpauldd** [comment]
@aliesbelik I think the checkver check failed because there is a special routine to get the current version (as discussed earlier in this thread)
when manully testet everything is okay:
```PowerShell
PS D:\Tools\scoop\buckets> scoop bucket add gigbotan_extras https://github.com/gigabotan/Extras.git
Checking repo... OK
The gigbotan_extras bucket was added successfully.
PS D:\Tools\scoop\buckets> cd .\gigbotan_extras\
PS D:\Tools\scoop\buckets\gigbotan_extras> git checkout opera-update-fix
Switched to a new branch 'opera-update-fix'
branch 'opera-update-fix' set up to track 'origin/opera-update-fix'.
PS D:\Tools\scoop\buckets\gigbotan_extras> .\bin\checkver.ps1 .\bucket\opera.json
opera: 106.0.4998.52 (scoop version is 104.0.4944.54) autoupdate available
PS D:\Tools\scoop\buckets\gigbotan_extras> .\bin\checkver.ps1 .\bucket\opera.json -u
opera: 106.0.4998.52 (scoop version is 104.0.4944.54) autoupdate available
Autoupdating opera
Searching hash for Opera_106.0.4998.52_Setup.exe in https://get.geo.opera.com/pub/opera/desktop/106.0.4998.52/win/Opera_106.0.4998.52_Setup.exe.sha256sum
Found: ff109164933428124bc0c9df96a97c949fd4d8562bfb678aa0ba0d2d74d98c57 using Extract Mode
Searching hash for Opera_106.0.4998.52_Setup_x64.exe in https://get.geo.opera.com/pub/opera/desktop/106.0.4998.52/win/Opera_106.0.4998.52_Setup_x64.exe.sha256sum
Found: e4ece354dff568fae1b00298b71201f63a89d11ac646d34259f50f7bcd519f92 using Extract Mode
Writing updated opera manifest
PS D:\Tools\scoop\buckets\gigbotan_extras>

```

**Comment by danielpauldd** [comment]
ah okay, forget it, the checkver task is ony checking if its the current version. 

Yes it is not up to date, because PR is 3months old and there are new version every few weeks. but will bumped up by github action, after merged.

**Pull request merged**